File: decompiled/4.0.2/be/ehealth/businessconnector/therlink/domain/requests/TherapeuticLinkComplete.java
package be.ehealth.businessconnector.therlink.domain.requests;

import be.ehealth.business.common.domain.Patient;
import be.ehealth.businessconnector.therlink.domain.HcParty;
import be.ehealth.businessconnector.therlink.domain.OperationContext;
import be.ehealth.businessconnector.therlink.domain.TherapeuticLink;
import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Iterator;
import java.util.List;
import org.apache.commons.lang.builder.ToStringBuilder;

public class TherapeuticLinkComplete extends TherapeuticLink {
   private static final long serialVersionUID = 1L;
   private List<OperationContext> operationContext;

   public TherapeuticLinkComplete(Patient patient, HcParty hcParty, String type, List<OperationContext> operationContext) {
      super(patient, hcParty, type);
      this.operationContext = operationContext;
   }

   public List<OperationContext> getOperationContext() {
      return this.operationContext;
   }

   public void setOperationContext(List<OperationContext> operationContext) {
      this.operationContext = operationContext;
   }

   public String toString() {
      ToStringBuilder builder = new ToStringBuilder(this);
      DateFormat df = new SimpleDateFormat("dd-MM-yyyy");
      builder.append("TherapeuticLink [HCP = ");
      builder.append(this.hcParty);
      builder.append(", Patient = ");
      builder.append(this.patient);
      if (this.startDate != null) {
         builder.append(", Startdate = ");
         builder.append(df.format(this.startDate.toDateMidnight()));
      }

      if (this.endDate != null) {
         builder.append(", Enddate = ");
         builder.append(df.format(this.endDate.toDateMidnight()));
      }

      builder.append(", Comment = ");
      builder.append(this.comment);
      builder.append(", Status = ");
      builder.append(this.status);
      builder.append(", OperationContexts = ");
      Iterator var3 = this.operationContext.iterator();

      while(var3.hasNext()) {
         OperationContext oc = (OperationContext)var3.next();
         builder.append("[");
         builder.append(oc.getOperation());
         builder.append("]");
      }

      builder.append("]");
      return builder.toString();
   }

   public int hashCode() {
      int prime = true;
      int result = super.hashCode();
      result = 31 * result + (this.operationContext == null ? 0 : this.operationContext.hashCode());
      return result;
   }

   public boolean equals(Object obj) {
      if (this == obj) {
         return true;
      } else if (!super.equals(obj)) {
         return false;
      } else if (this.getClass() != obj.getClass()) {
         return false;
      } else {
         TherapeuticLinkComplete other = (TherapeuticLinkComplete)obj;
         if (this.operationContext == null) {
            if (other.operationContext != null) {
               return false;
            }
         } else if (!this.operationContext.equals(other.operationContext)) {
            return false;
         }

         return true;
      }
   }
}

File: src/main/kotlin/org/taktik/connector/business/therlink/domain/requests/TherapeuticLinkComplete.kt
package org.taktik.connector.business.therlink.domain.requests

import org.taktik.connector.business.common.domain.Patient
import org.taktik.connector.business.therlink.domain.HcParty
import org.taktik.connector.business.therlink.domain.OperationContext
import org.taktik.connector.business.therlink.domain.TherapeuticLink
import java.text.SimpleDateFormat
import org.apache.commons.lang.builder.ToStringBuilder

class TherapeuticLinkComplete(
    patient: Patient,
    hcParty: HcParty,
    type: String,
    var operationContext: List<OperationContext>?
) : TherapeuticLink(patient, hcParty, type) {

    override fun toString(): String {
        val builder = ToStringBuilder(this)
        val df = SimpleDateFormat("dd-MM-yyyy")
        builder.append("TherapeuticLink [HCP = ")
        builder.append(this.hcParty)
        builder.append(", Patient = ")
        builder.append(this.patient)
        builder.append(", Startdate = ")
        this.startDate?.let { builder.append(df.format(it.toDateMidnight())) }
        this.endDate?.let { builder.append(df.format(it.toDateMidnight())) }

        builder.append(", Comment = ")
        builder.append(this.comment)
        builder.append(", Status = ")
        builder.append(this.status)
        builder.append(", OperationContexts = ")
        val `i$` = this.operationContext!!.iterator()

        while (`i$`.hasNext()) {
            builder.append("[")
            builder.append(`i$`.next().operation)
            builder.append("]")
        }

        builder.append("]")
        return builder.toString()
    }

    override fun hashCode(): Int {
        var result = super.hashCode()
        result = 31 * result + if (this.operationContext == null) 0 else this.operationContext!!.hashCode()
        return result
    }

    override fun equals(obj: Any?): Boolean {
        if (this === obj) {
            return true
        } else if (!super.equals(obj)) {
            return false
        } else if (this.javaClass != obj!!.javaClass) {
            return false
        } else {
            val other = obj as TherapeuticLinkComplete?
            if (this.operationContext == null) {
                if (other!!.operationContext != null) {
                    return false
                }
            } else if (this.operationContext != other!!.operationContext) {
                return false
            }

            return true
        }
    }

    companion object {
        private val serialVersionUID = 1L
    }
}
